mapzen-places-api/src/main/java/com/mapzen/places/api/internal/PeliasPlaceDetailFetcher.java

```diff
@@ -0,0 +1,81 @@
+package com.mapzen.places.api.internal;
+
+import com.mapzen.pelias.Pelias;
+import com.mapzen.pelias.gson.Feature;
+import com.mapzen.pelias.gson.Result;
+import com.mapzen.places.api.LatLng;
+import com.mapzen.places.api.LatLngBounds;
+import com.mapzen.places.api.Place;
+import com.mapzen.tangram.LngLat;
+
+import android.net.Uri;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static com.mapzen.places.api.internal.PlacePickerPresenterImpl.PROPERTY_NAME;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Detail fetcher which uses pelias as the backing data source.
+ */
+public class PeliasPlaceDetailFetcher implements PlaceDetailFetcher {
+
+  Pelias pelias;
+  Feature feature;
+  LngLat coordinates;
+  Map<String, String> properties;
+
+  /**
+   * Constructs a new object.
+   */
+  public PeliasPlaceDetailFetcher() {
+    pelias = new Pelias();
+  }
+
+  @Override public void fetchDetails(LngLat coordinates, final Map<String, String> properties,
+      final OnPlaceDetailsFetchedListener listener) {
+    this.coordinates = coordinates;
+    this.properties = properties;
+    pelias.reverse(coordinates.latitude, coordinates.longitude, new Callback<Result>() {
+          @Override public void onResponse(Call<Result> call, Response<Result> response) {
+            String title = properties.get(PROPERTY_NAME);
+            for (Feature feature : response.body().getFeatures()) {
+              if (feature.properties.name.equals(title)) {
+                PeliasPlaceDetailFetcher.this.feature = feature;
+                String label = feature.properties.label;
+                label = label.replace(title + ",", "").trim();
+                listener.onPlaceDetailsFetched(title + "\n" + label);
```

Comment:
Due to the volatility of OSM IDs and differences in data pipeline for search and tiles it seems this is our best bet for now.

One alternative would be to perform a forward geocode on the name of the venue and source but this still is not guaranteed to produce an exact match.

Let's keep as is for now and hopefully in the future we'll be able to take advantage of a shared data pool/pipeline with IDs normalized across search and tiles.